### ISSUE TITLE
NotReadableError: Could not start video source

### ISSUE THREAD

**Issue by glalloue** [zxing-js/ngx-scanner#241]
I updated ngx-scanner from 1.5.3 to 2.0.1 and modified call of zxing-scanner.
But when i run my web app, i receive this message on logs : **NotReadableError: Could not start video source**

![image](https://user-images.githubusercontent.com/10628710/64120259-0391f480-cd9c-11e9-9a8e-bcb0309e1cf9.png)

My code : 
scan-qrcode.page.html : 
```
<zxing-scanner #scanner
    [enable]="true"
    [class.hidden]="!isScannerAllowed"
    [(device)]="selectedCamera"
    [autofocusEnabled]="true"
    (scanSuccess)="handleQrcode($event)"
    [formats]="formatsEnabled">
</zxing-scanner>
```
scan-qrcode.page.ts : 
```
export class ScanQrcodePage implements OnInit, OnDestroy {
  [...]
  public ngOnInit() {
    this.subscriptionCameras = this.scanner.camerasFound.subscribe((cameras: MediaDeviceInfo[]) => {
      if (cameras && cameras.length > 0) {
        this.hasCamera = true;
        this.selectedCamera = cameras.find((c) => c.label.includes('back')) || cameras[0];
        //this.scanner.startScan(this.selectedCamera);
      }
    });
  }
}
```

Informations : 
 - Browser : Edge Chromium 77.0.235.5 
 - Angular 8.1.3
- ngx-scanner : 2.0.1

**Comment by glalloue** [comment]
If i move  : 
`[(device)]="selectedCamera"` 
from HTML to typescript code : 
`this.scanner.device = this.selectedCamera;`
it seems to be working.
=> the documentation is not very precise about the obligation to set "device".
=> it's working, but i have always the same error with a warning too : 
![image](https://user-images.githubusercontent.com/10628710/64121608-a009c600-cd9f-11e9-9ba2-6451c3e03e8c.png)

**Comment by odahcam** [comment]
Sorry, I'm not familiar with this error or obligation, device shouldn't be required as well. 🤔

**Comment by tilakdewangan-gep** [comment]
I am also facing the same issue and this is very much consistent.

**Comment by jacobthomasdpr** [comment]
I am getting the same error when testing on Samsung Galaxy note 10,  on chrome.
I don't get this error on Pixel XL,  or IOS.  The demo from https://www.npmjs.com/package/@zxing/ngx-scanner  was tested. 

Is there anything device specific that might be causing it?

there seems to be a similar problem here 
https://github.com/philnash/mediadevices-camera-selection/issues/1

Thanks

**Comment by odahcam** [comment]
Well, maybe it can be related at all: https://github.com/philnash/mediadevices-camera-selection/issues/1#issuecomment-569465918

**Comment by splowman-FormFast** [comment]
I am doing some testing on my local machine and I run into this issue when I purposely disable one of the cameras (When I don't disable it, it works fine).  The camera still shows up on the camerasFound event when I would think it wouldn't.  I am running this on a Windows 10 machine.

**Comment by mishraswapn** [comment]
> If i move :
> `[(device)]="selectedCamera"`
> from HTML to typescript code :
> `this.scanner.device = this.selectedCamera;`
> it seems to be working.
> => the documentation is not very precise about the obligation to set "device".
> => it's working, but i have always the same error with a new warning :
> ![image](https://user-images.githubusercontent.com/10628710/64121608-a009c600-cd9f-11e9-9ba2-6451c3e03e8c.png)



> I updated ngx-scanner from 1.5.3 to 2.0.1 and modified call of zxing-scanner.
> But when i run my web app, i receive this message on logs : **NotReadableError: Could not start video source**
> 
> ![image](https://user-images.githubusercontent.com/10628710/64120259-0391f480-cd9c-11e9-9a8e-bcb0309e1cf9.png)
> 
> My code :
> scan-qrcode.page.html :
> 
> ```
> <zxing-scanner #scanner
>     [enable]="true"
>     [class.hidden]="!isScannerAllowed"
>     [(device)]="selectedCamera"
>     [autofocusEnabled]="true"
>     (scanSuccess)="handleQrcode($event)"
>     [formats]="formatsEnabled">
> </zxing-scanner>
> ```
> 
> scan-qrcode.page.ts :
> 
> ```
> export class ScanQrcodePage implements OnInit, OnDestroy {
>   [...]
>   public ngOnInit() {
>     this.subscriptionCameras = this.scanner.camerasFound.subscribe((cameras: MediaDeviceInfo[]) => {
>       if (cameras && cameras.length > 0) {
>         this.hasCamera = true;
>         this.selectedCamera = cameras.find((c) => c.label.includes('back')) || cameras[0];
>         //this.scanner.startScan(this.selectedCamera);
>       }
>     });
>   }
> }
> ```
> 
> Informations :
> 
> * Browser : Edge Chromium 77.0.235.5
> * Angular 8.1.3
> * ngx-scanner : 2.0.1

were you able to make this work? strange thing is I was using this successfully on a windows 10 with latest Chrome, but not able to use it on another machine with win10 and Chrome. Getting this error, checked all the permissions are granted, but not able to start the feed.

**Comment by mding5692** [comment]
Could be related to this: https://github.com/twilio/twilio-video.js/issues/325

**Comment by odahcam** [comment]
Sorry, this is too outdated now. I ask you to update the library to the latest version available.

**Comment by loic-parent** [comment]
Hello,

I encounter this issue with this package.

This error message appears only on some devices (not all):
![image](https://github.com/user-attachments/assets/3d39bea6-5b97-4bf5-bda0-17104ccc334e)

(I am using Ionic with Angular and the package @zxing/ngx-scanner with the latest version).

The permission is ok, I tried the method `restart` and `tryHarder` and I also tried to get and set the devices but, unfortunately, without any success.

**Is there a way to fix it?**

_Thanks for your help,
Loïc_
